scenes/menu/main_menu.gd

```diff
@@ -430,11 +442,18 @@ func _on_modal_list_observe_match_pressed(row_index):
 	NetworkManager.observe_room(player_name, room_name)
 	update_buttons(true)
 
-func _on_view_replay_button_pressed():
-	if replay_data_box.text:
-		var replay_data = JSON.parse_string(replay_data_box.text)
-		_on_observe_game_started(replay_data)
+func _on_load_replay_button_pressed():
+	if OS.has_feature("web"):
+		window.input.click()
+	else:
+		file_dialog.visible = true
 
 func _on_settings_button_pressed():
 	settings_window.visible = true
 
+func load_replay(data):
+	var replay_data = JSON.parse_string(data[0])
+	_on_observe_game_started(replay_data, true)
+
+func _on_file_dialog_file_selected(path):
+	load_replay([FileAccess.get_file_as_string(path)])
```

Comment:
Did you test canceling the dialog to make sure that works?

Reply:
Yeah, this only gets called when a file is confirmed, and clicking the Load Replay button only pops out the dialog, so nothing happens if you cancel.